charts/dex/templates/ingress.yaml

```diff
@@ -1,6 +1,6 @@
 {{- if .Values.ingress.enabled -}}
 {{- $fullName := include "dex.fullname" . -}}
-{{- $svcPort := .Values.service.ports.http.port -}}
+{{- $svcPort := .Values.https.enabled | ternary .Values.service.ports.http.port .Values.service.port.https.port}}
```

Comment:
I think http and https are switched here; I believe https should come first, i.e. when .Values.https.enabled is true.
See a cilium example https://github.com/cilium/cilium/blob/687e4f8f403aa75f3e1e42005db51b29fdd9e5ef/install/kubernetes/cilium/templates/hubble/peer-service.yaml#L23
Also, I think you need to keep the dash at the end of the expression to prevent a new line. (You need to keep `-}}` instead of changing it to `}}`)